data-service/src/main/java/org/ihtsdo/buildcloud/service/ProductServiceImpl.java

```diff
@@ -296,8 +304,19 @@ private void updateProductBuildConfiguration(final Map<String, String> newProper
         if (newPropertyValues.containsKey(USE_CLASSIFIER_PRECONDITION_CHECKS)) {
             configuration.setUseClassifierPreConditionChecks(TRUE.equals(newPropertyValues.get(USE_CLASSIFIER_PRECONDITION_CHECKS)));
         }
-        if (newPropertyValues.containsKey(DEFAULT_BRANCH_PATH)) {
-            configuration.setDefaultBranchPath(newPropertyValues.get(DEFAULT_BRANCH_PATH));
+        if (newPropertyValues.containsKey(DEFAULT_BRANCH_PATH) && configuration.getDefaultBranchPath() != newPropertyValues.get(DEFAULT_BRANCH_PATH).toUpperCase()) {
+            String newDefaultBranchPath = newPropertyValues.get(DEFAULT_BRANCH_PATH);
+            if (!StringUtils.isEmpty(newDefaultBranchPath)) {
+                List<CodeSystem> codeSystems = termServerService.getCodeSystems();
+                CodeSystem codeSystem = codeSystems.stream()
+                        .filter(c -> product.getReleaseCenter().getCodeSystem().equals(c.getShortName()))
+                        .findAny()
+                        .orElse(null);
+                if (!newDefaultBranchPath.toUpperCase().startsWith(codeSystem.getBranchPath())) {
```

Comment:
Branch paths are case sensitive, I don't think we should convert the case when validating. If the branch has the wrong case it should fail.

Reply:
Updated